client/src/main.rs

```diff
@@ -54,13 +54,18 @@ fn main() -> std::io::Result<()> {
     crash_handler::setup_panic_handler();
     let args = Args::from_args();
 
+    if args.port == 0 {
+        println!("Port number cannot be zero, zero is reserved for random port.");
+        return Ok(());
+    }
+
     let (commands, alias_to_cmd) = get_commands(args.faucet_account_file.is_some());
 
     let faucet_account_file = args.faucet_account_file.unwrap_or_else(|| "".to_string());
 
     let mut client_proxy = ClientProxy::new(
         &args.host,
-        &args.port,
+        args.port,
```

Comment:
And then since `args.port` would be of type `NonZeroU16` you'd need to call the get method to get the u16 out of it: [`args.port.get()`](https://doc.rust-lang.org/std/num/struct.NonZeroU16.html#method.get).